crates/admin/src/cluster_controller/scheduler.rs

```diff
@@ -45,9 +47,22 @@ pub enum Error {
     Shutdown(#[from] ShutdownError),
 }
 
+enum WriteOrFetch<T> {
+    Write(T),
+    Fetch(T),
+}
```

Comment:
I like this a lot! Maybe a more intuitive naming here would be:

```rust
enum UpdateOutcome<T> {
    Written(T),
    NewerVersionFound(T),
}
```

then the write helper signature can be `try_update(T) -> Result<UpdateOutcome<T>>`

Reply:
Good idea. Will change it.